ft-fine-tune-bert/fine-tune-unittests.py

```diff
@@ -0,0 +1,118 @@
+            ],
+         ]
+
+        with open("6_genre_clean_training_data_small_unittests.txt", "w") as output:
+            for row in self.train_data:
+                for col in row: 
+                    output.write(str(col) + '\n')
+
+        with open("6_genre_eval_data_small_unittests.txt", "w") as output:
+            for row in self.eval_data:
+                for col in row:
+                    output.write(str(col) + '\n')
+
+
+    def assertIsFile(self, path):
+        if not pl.Path(path).resolve().is_file():
+            raise AssertionError("File does not exist: %s" % str(path))
+    
+    def assertFileExist(self, path):
+        if not pl.Path(path):
+            raise ValueError(
+                "Cannot do evaluation without an evaluation data file. Either supply a file to --eval_data_file "
+                "or remove the --do_eval argument."
+            )
+    def assertDirExist(self, path):
+        if (
+            os.path.exists(path)
+        ):
+            raise ValueError(
+                f"Output directory ({path}) already exists and is not empty. Use --overwrite_output_dir to overcome."
+            )
+
+    # Test 1
+    def test_data_paths(self):
+        """Checks if input file existing"""
+        file = pl.Path("./6_genre_clean_training_data_small.txt")
+        self.assertIsFile(file)
+
+    # Test 2
+    def test_data_paths(self):
+        """Checks if input file existing"""
+        file = pl.Path("./6_genre_eval_data_small.txt")
+        self.assertIsFile(file)
+
+    # Test 3
+    def test_eval_exit(self):
+        """Checks if Eval file provided"""
+        file = pl.Path("./6_genre_eval_data_small.txt")
+        self.assertFileExist(file)
+
+    # Test 4
+    def test_outputdir_exist(self):
+        """Checks if output_dir existing"""
+        file = pl.Path("./story_generator_checkpoint_" + "bert-base-uncased")
+        self.assertDirExist(file)
+
+    # Test 5
+    def test_return_type(self):
+        """Checks if the function returns the type of transformers.data.datasets.language_modeling.LineByLineTextDataset"""
+        data_args = DataTrainingArguments(
+            train_data_file='6_genre_clean_training_data_small_unittests.txt',
+            eval_data_file='6_genre_eval_data_small_unittests.txt',
+            line_by_line=True,
+            block_size=256,
+
+            overwrite_cache=True,
+        )
+        tokenizer = AutoTokenizer.from_pretrained("bert-base-uncased")
+        returned_lists_trainy = get_dataset(data_args, tokenizer=tokenizer)
+        returned_lists_evaly = get_dataset(data_args, tokenizer=tokenizer, evaluate=True)
+        self.assertIsInstance(returned_lists_trainy, transformers.data.datasets.language_modeling.LineByLineTextDataset)
```

Comment:
Return type testing is not enough. 

Need to add test cases to actual training / fine tune performance.

Reply:
Add unittests to check performance metrics perplexity.